src/main/java/org/broadinstitute/hellbender/tools/walkers/annotator/VariantAnnotatorEngine.java

```diff
@@ -0,0 +1,296 @@
+package org.broadinstitute.hellbender.tools.walkers.annotator;
+
+import com.google.common.collect.Sets;
+import htsjdk.variant.variantcontext.*;
+import htsjdk.variant.vcf.*;
+import org.broadinstitute.hellbender.cmdline.ClassFinder;
+import org.broadinstitute.hellbender.engine.FeatureContext;
+import org.broadinstitute.hellbender.engine.FeatureInput;
+import org.broadinstitute.hellbender.engine.ReferenceContext;
+import org.broadinstitute.hellbender.exceptions.UserException.BadArgumentValue;
+import org.broadinstitute.hellbender.utils.Utils;
+import org.broadinstitute.hellbender.utils.genotyper.PerReadAlleleLikelihoodMap;
+import org.reflections.ReflectionUtils;
+
+import java.util.*;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+/**
+ * The class responsible for computing annotations for variants.
+ */
+public final class VariantAnnotatorEngine {
+    private final List<InfoFieldAnnotation> infoAnnotations;
+    private final List<GenotypeAnnotation> genotypeAnnotations;
+
+    private final VariantOverlapAnnotator variantOverlapAnnotator;
+
+    private VariantAnnotatorEngine(final AnnotationManager annots, final FeatureInput<VariantContext> dbSNPInput){
+        infoAnnotations = annots.createInfoFieldAnnotations();
+        genotypeAnnotations = annots.createGenotypeAnnotations();
+        variantOverlapAnnotator = initializeDBs(dbSNPInput);
+    }
+
+    /**
+     * Makes the engine for all known annotation types (minus the excluded ones).
+     * @param annotationsToExclude list of annotations to exclude
```

Comment:
the convention is (and I think we stick to it) that if the doc does not mention that something can be null, then it cannot be null. I added `(pass an empty list to indicate that there are no exclusions)`